### PR TITLE
Prepare 6.0.0 changelog

### PR DESCRIPTION
**Pull request by domcleal**
[ci skip]

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,41 @@
...
+    * Add server_port, server_ssl_port parameters to change Apache vhost ports
+    * Rename environment parameter to rails_env to fix compatibility with data
+      bindings
+* Other changes and fixes:
+    * node.rb: skip facts upload when facts file is missing, retrieves ENC
+      output anyway
+    * node.rb: improve logging for empty facts and failed fact uploads
+    * Change reports upload to use new config_reports API
+    * Change Yum GPG key URLs to HTTPS
+    * Fix missing default parameters for strict variables compatibility,
```
Comment (mmoll) [inline CHANGELOG.md]: #476 worked that around (at least for that one place), so requiring 4.6 should be ok, like it's stated in line 24?

Comment (domcleal) [inline CHANGELOG.md]: It only worked around it when strict variables is disabled on Puppet 3.6.0-3.7.4, unfortunately it'll still raise an error on the pick() line if strict vars is enabled on those versions. I was trying to say here that you can use the module with 3.6.0 or higher, but only 3.7.5 or higher if you have strict vars enabled.

Comment (mmoll) [inline CHANGELOG.md]: ah, I see... :+1:

**Comment by mmoll** [comment]
merged, thanks @domcleal!

**Pull request merged**